splash/browser_tab.py

```diff
@@ -299,6 +301,16 @@ def go(self, url, callback, errback, baseurl=None, http_method='GET',
         if body is not None:
             body = to_bytes(body)
 
+        if headers is not None:
+            # headers can be list or dict, keys can be bytes or str, HTTP specs
+            # say headers are case-insensitive, so they can be uppercase or lowercase
+            if not isinstance(headers, dict):
+                headers = dict(headers)
+            headers = {to_bytes(k.lower()): to_bytes(v) for k, v in headers.items()}
```

Comment:
> Wouldn't this make all headers lower-case?

yeah it will make all headers case insensitive. I thought that it would be easier for us to work with headers in consistent case, it is irritating to account for case when fetching headers from dictionary. 

But you're right that perhaps users care about case they pass so we should not change it, so I will update this.